error[E0277]: the trait bound `Q: rocket::request::FromParam<'_>` is not satisfied
 --> $DIR/route-type-errors.rs:6:7
  |
6 | fn f0(foo: Q) {} //~ ERROR FromParam
  |       ^^^^^^ the trait `rocket::request::FromParam<'_>` is not implemented for `Q`

error[E0277]: the trait bound `Q: rocket::request::FromSegments<'_>` is not satisfied
 --> $DIR/route-type-errors.rs:9:7
  |
9 | fn f1(foo: Q) {} //~ ERROR FromSegments
  |       ^^^^^^ the trait `rocket::request::FromSegments<'_>` is not implemented for `Q`

error[E0277]: the trait bound `Q: rocket::request::FromFormValue<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:12:7
   |
12 | fn f2(foo: Q) {} //~ ERROR FromFormValue
   |       ^^^^^^ the trait `rocket::request::FromFormValue<'_>` is not implemented for `Q`

error[E0277]: the trait bound `Q: rocket::request::FromQuery<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:15:7
   |
15 | fn f3(foo: Q) {} //~ ERROR FromQuery
   |       ^^^^^^ the trait `rocket::request::FromQuery<'_>` is not implemented for `Q`

error[E0277]: the trait bound `Q: rocket::data::FromData` is not satisfied
  --> $DIR/route-type-errors.rs:18:7
   |
18 | fn f4(foo: Q) {} //~ ERROR FromData
   |       ^^^^^^ the trait `rocket::data::FromData` is not implemented for `Q`
   |
   = note: required because of the requirements on the impl of `rocket::data::FromTransformedData<'_>` for `Q`

error[E0277]: the trait bound `Q: rocket::request::FromRequest<'_, '_>` is not satisfied
  --> $DIR/route-type-errors.rs:21:7
   |
21 | fn f5(a: Q, foo: Q) {}
   |       ^^^^ the trait `rocket::request::FromRequest<'_, '_>` is not implemented for `Q`

error[E0277]: the trait bound `Q: rocket::request::FromParam<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:21:13
   |
21 | fn f5(a: Q, foo: Q) {}
   |             ^^^^^^ the trait `rocket::request::FromParam<'_>` is not implemented for `Q`

error[E0277]: the trait bound `Q: rocket::request::FromRequest<'_, '_>` is not satisfied
  --> $DIR/route-type-errors.rs:26:7
   |
26 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |       ^^^^ the trait `rocket::request::FromRequest<'_, '_>` is not implemented for `Q`

error[E0277]: the trait bound `Q: rocket::request::FromParam<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:26:13
   |
26 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |             ^^^^^^ the trait `rocket::request::FromParam<'_>` is not implemented for `Q`

error[E0277]: the trait bound `Q: rocket::request::FromParam<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:26:34
   |
26 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |                                  ^^^^^^ the trait `rocket::request::FromParam<'_>` is not implemented for `Q`

error: aborting due to 10 previous errors

For more information about this error, try `rustc --explain E0277`.
