error: path parameters cannot be ignored
  --> $DIR/typed-uris-bad-params.rs:77:37
   |
77 |     uri!(optionals: id = 10, name = _);
   |                                     ^

error: path parameters cannot be ignored
  --> $DIR/typed-uris-bad-params.rs:74:26
   |
74 |     uri!(optionals: id = _, name = "bob".into());
   |                          ^

error: invalid parameters for `has_two` route uri
  --> $DIR/typed-uris-bad-params.rs:70:19
   |
70 |     uri!(has_two: id = 100, cookies = "hi"); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32, name: String
   = help: missing parameter: `name`
help: unknown parameter: `cookies`
  --> $DIR/typed-uris-bad-params.rs:70:29
   |
70 |     uri!(has_two: id = 100, cookies = "hi"); //~ ERROR invalid parameters
   |                             ^^^^^^^

error: invalid parameters for `has_two` route uri
  --> $DIR/typed-uris-bad-params.rs:65:19
   |
65 |     uri!(has_two: cookies = "hi", id = 100, id = 10, id = 10); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32, name: String
   = help: missing parameter: `name`
help: unknown parameter: `cookies`
  --> $DIR/typed-uris-bad-params.rs:65:19
   |
65 |     uri!(has_two: cookies = "hi", id = 100, id = 10, id = 10); //~ ERROR invalid parameters
   |                   ^^^^^^^
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:65:45
   |
65 |     uri!(has_two: cookies = "hi", id = 100, id = 10, id = 10); //~ ERROR invalid parameters
   |                                             ^^       ^^

error: invalid parameters for `has_two` route uri
  --> $DIR/typed-uris-bad-params.rs:62:19
   |
62 |     uri!(has_two: name = "hi"); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32, name: String
   = help: missing parameter: `id`

error: invalid parameters for `has_two` route uri
  --> $DIR/typed-uris-bad-params.rs:58:19
   |
58 |     uri!(has_two: id = 100, id = 100, ); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32, name: String
   = help: missing parameter: `name`
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:58:29
   |
58 |     uri!(has_two: id = 100, id = 100, ); //~ ERROR invalid parameters
   |                             ^^

error: invalid parameters for `has_one_guarded` route uri
  --> $DIR/typed-uris-bad-params.rs:55:27
   |
55 |     uri!(has_one_guarded: id = 100, cookies = "hi"); //~ ERROR invalid parameters
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameter: `cookies`
  --> $DIR/typed-uris-bad-params.rs:55:37
   |
55 |     uri!(has_one_guarded: id = 100, cookies = "hi"); //~ ERROR invalid parameters
   |                                     ^^^^^^^

error: invalid parameters for `has_one_guarded` route uri
  --> $DIR/typed-uris-bad-params.rs:52:27
   |
52 |     uri!(has_one_guarded: cookies = "hi", id = 100); //~ ERROR invalid parameters
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameter: `cookies`
  --> $DIR/typed-uris-bad-params.rs:52:27
   |
52 |     uri!(has_one_guarded: cookies = "hi", id = 100); //~ ERROR invalid parameters
   |                           ^^^^^^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:48:19
   |
48 |     uri!(has_one: name = "hi"); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
   = help: missing parameter: `id`
help: unknown parameter: `name`
  --> $DIR/typed-uris-bad-params.rs:48:19
   |
48 |     uri!(has_one: name = "hi"); //~ ERROR invalid parameters
   |                   ^^^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:45:19
   |
45 |     uri!(has_one: id = 100, id = 100, ); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:45:29
   |
45 |     uri!(has_one: id = 100, id = 100, ); //~ ERROR invalid parameters
   |                             ^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:42:19
   |
42 |     uri!(has_one: id = 100, id = 100); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:42:29
   |
42 |     uri!(has_one: id = 100, id = 100); //~ ERROR invalid parameters
   |                             ^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:38:19
   |
38 |     uri!(has_one: name = 100, age = 50, id = 100, id = 50); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameters: `name`, `age`
  --> $DIR/typed-uris-bad-params.rs:38:19
   |
38 |     uri!(has_one: name = 100, age = 50, id = 100, id = 50); //~ ERROR invalid parameters
   |                   ^^^^        ^^^
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:38:51
   |
38 |     uri!(has_one: name = 100, age = 50, id = 100, id = 50); //~ ERROR invalid parameters
   |                                                   ^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:35:19
   |
35 |     uri!(has_one: name = 100, age = 50, id = 100); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameters: `name`, `age`
  --> $DIR/typed-uris-bad-params.rs:35:19
   |
35 |     uri!(has_one: name = 100, age = 50, id = 100); //~ ERROR invalid parameters
   |                   ^^^^        ^^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:32:19
   |
32 |     uri!(has_one: name = 100, id = 100); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameter: `name`
  --> $DIR/typed-uris-bad-params.rs:32:19
   |
32 |     uri!(has_one: name = 100, id = 100); //~ ERROR invalid parameters
   |                   ^^^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:29:19
   |
29 |     uri!(has_one: id = 100, name = "hi"); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameter: `name`
  --> $DIR/typed-uris-bad-params.rs:29:29
   |
29 |     uri!(has_one: id = 100, name = "hi"); //~ ERROR invalid parameters
   |                             ^^^^

error: `has_two` route uri expects 2 parameters but 1 was supplied
  --> $DIR/typed-uris-bad-params.rs:27:19
   |
27 |     uri!(has_two: 10); //~ ERROR expects 2 parameters but 1
   |                   ^^
   |
   = note: expected parameters: id: i32, name: String

error: `has_two` route uri expects 2 parameters but 3 were supplied
  --> $DIR/typed-uris-bad-params.rs:26:19
   |
26 |     uri!(has_two: 10, "hi", "there"); //~ ERROR expects 2 parameters but 3
   |                   ^^^^^^^^^^^^^^^^^
   |
   = note: expected parameters: id: i32, name: String

error: `has_one_guarded` route uri expects 1 parameter but 2 were supplied
  --> $DIR/typed-uris-bad-params.rs:24:27
   |
24 |     uri!(has_one_guarded: "hi", 100); //~ ERROR expects 1 parameter but 2
   |                           ^^^^^^^^^
   |
   = note: expected parameter: id: i32

error: `has_one` route uri expects 1 parameter but 2 were supplied
  --> $DIR/typed-uris-bad-params.rs:23:19
   |
23 |     uri!(has_one: "Hello", 23, ); //~ ERROR expects 1 parameter but 2
   |                   ^^^^^^^^^^^^
   |
   = note: expected parameter: id: i32

error: `has_one` route uri expects 1 parameter but 2 were supplied
  --> $DIR/typed-uris-bad-params.rs:22:19
   |
22 |     uri!(has_one: 1, 23); //~ ERROR expects 1 parameter but 2
   |                   ^^^^^
   |
   = note: expected parameter: id: i32

error: `has_one` route uri expects 1 parameter but 0 were supplied
  --> $DIR/typed-uris-bad-params.rs:20:10
   |
20 |     uri!(has_one); //~ ERROR expects 1 parameter but 0
   |          ^^^^^^^
   |
   = note: expected parameter: id: i32

error: aborting due to 21 previous errors

